cypress/integration/homepage/home_page_latest_message_spec.js

```diff
@@ -1,44 +1,70 @@
 import { ImageDisplayValidator } from '../../Contentful/ImageDisplayValidator';
+import { RequestFilter } from '../../Analytics/RequestFilter';
+import { amplitude } from '../../fixtures/event_filters';
 import { MessageQueryManager } from 'crds-cypress-contentful';
 
-describe('Testing the Current Message on the Homepage:', function () {
+describe('Tests the Current Message on the Homepage', function () {
   let currentMessage;
-  let relativeAutoplayURL;
-  before(function () {
+  let requestFilter;
+
+  before(() => {
+    //Fetch Current Message
     const mqm = new MessageQueryManager();
     mqm.getSingleEntry(mqm.query.latestMessage).then(message => {
       currentMessage = message;
-      currentMessage.getURL().then(url => {
-        relativeAutoplayURL = url.autoplay.relative;
-      });
     });
 
+    //Setup capture for events
+    cy.server();
+    requestFilter = new RequestFilter(amplitude.isVideoStarted);
+    cy.route({
+      method: 'POST',
+      url: 'api.amplitude.com',
+      onResponse: (xhr) => {
+        requestFilter.keepMatch(xhr.request);
+      }
+    });
+
+    //Navigate
     cy.ignorePropertyUndefinedTypeError();
     cy.visit('/');
   });
 
-  it('Current Message title, description, and image should match Contentful', function () {
-    cy.get('[data-automation-id="message-title"]').as('title');
-    cy.get('@title').text().should('contain', currentMessage.title.text);
-    cy.get('@title').should('have.attr', 'href', relativeAutoplayURL);
+  it('Checks title, image, and "View latest now" button have correct link', () => {
```

Comment:
question: does Cypress reuse the session in a way that might make it so you can break this test up?

have a
`Checks title has correct link`
`Checks image has the correct link`
`Checks "View Latest now" button has correct link`

Then you end up with smaller tests that might be more readable and more debuggable.

Reply:
Unfortunately no, which is why you might notice me bundling more checks into a single test. Their before/after each processes run after each "it" block and seem to be pretty process heavy. A spec file similar to this one took 17s if tests were broken up into really small chunks but only 2s when more assertions were bundled together.
Cypress wants you to test multiple assertions in a single test, they even have an entry in their best practices section [here](https://docs.cypress.io/guides/references/best-practices.html#Creating-%E2%80%9Ctiny%E2%80%9D-tests-with-a-single-assertion). The tl;dr is Cypress tests aren't unit tests, so shouldn't be structured in the same way.

Reply:
gross. Understood